### PR TITLE
Infinite scrolling enhancements and profile button

### PR DESCRIPTION
**Pull request by Reddiepoint**
Cleaned up and simplified the infinite scrolling function and added the profile button next to UCP button.

### CODE REVIEW

File: cs-rin-ru-enhanced-mod.user.js [hunk 1]
```diff
@@ -367,7 +359,10 @@ function dynamicFunction(data) {
     //Call every 60seconds as well as when using infinite scroll
     $("#datebar .gensmall+ .gensmall").html($("#datebar .gensmall+ .gensmall", data).html()); //Time
     $("#wrapcentre > .tablebg").last().html($("#wrapcentre > .tablebg", data).last().html()); //Users
-    $("#menubar > table:nth-child(3) > tbody > tr > td:nth-child(1) > a:nth-child(2)").html($("#menubar > table:nth-child(3) > tbody > tr > td:nth-child(1) > a:nth-child(2)", data).html()); //Message
+    const html = $("#menubar > table:nth-child(3) > tbody > tr > td:nth-child(1) > a:nth-child(2)", data).html();
+    if ($(html)[0].src === 'https://cs.rin.ru/forum/styles/rinDark/theme/images/icon_mini_message.gif') {
```
Comment (SubZeroPL) [inline cs-rin-ru-enhanced-mod.user.js:363]: This will not work on Tor, you should use FORUM_BASE_URL in links

**Review by SubZeroPL — CHANGES_REQUESTED** [review]
Added some comments

**Comment by Reddiepoint** [comment]
Sorry for that, I forgot about Tor :(
I've addressed the changes now.

**Review by SubZeroPL — APPROVED** [review]
Looks fine for me

**Pull request merged**